src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.macios.cs

```diff
@@ -10,6 +10,7 @@ namespace CommunityToolkit.Maui.Core.Views;
 public class MauiPopup : UIViewController
 {
 	readonly IMauiContext mauiContext;
+	UIView overlayView;
```

Comment:
@brminnick , Sorry, I overlooked WeakReference.
I would like to change the implementation to use WeakReference.

Reply:
@brminnick , After changing to use WeakReference, overlayView may be destroyed at an unintended 
timing when Popup is repeatedly show and close. Does this mean it has been disposed of by the 
garbage collector?

Reply:
Off of the top of my head, I don't think that's too much of a concern because we disconnect the Popup's Handler and dispose of it when `IPopup.Close()` is called: 
https://github.com/CommunityToolkit/Maui/blob/0ab0fe22f4e701d1f463e243f85c2ddb07354a6a/src/CommunityToolkit.Maui.Core/Handlers/Popup/PopupHandler.macios.cs#L14-L25

I haven't done any testing, though. If you found a scenario where `overlayView` is disposed before Popup is disposed, let me know!

Reply:
@brminnick , Thank you for your reply.

I made the changes below.

[src\CommunityToolkit.Maui.Core\Views\Popup\MauiPopup.macios.cs]

    readonly WeakReference<UIView> overlayViewReference = new(new UIView());

    internal UIView OverlayView
    {
        get => overlayViewReference.TryGetTarget(out var overlayView) ? overlayView : throw new ObjectDisposedException(nameof(OverlayView));
        set => overlayViewReference.SetTarget(value);
    }

    public override void ViewWillDisappear(bool animated)
    {
        OverlayView.RemoveFromSuperview();
        base.ViewWillDisappear(animated);
    }

    public override void ViewWillTransitionToSize(CGSize toSize, IUIViewControllerTransitionCoordinator coordinator)
    {
        coordinator.AnimateAlongsideTransition((IUIViewControllerTransitionCoordinatorContext obj) =>
        {
            // Before screen rotate
            if (ViewController?.View is UIView view)
            {
                OverlayView.Frame = new CGRect(view.Frame.X, view.Frame.Y, view.Frame.Width, view.Frame.Height);
            }
        }, (IUIViewControllerTransitionCoordinatorContext obj) =>
        {
            // After screen rotate
            if (VirtualView is not null)
            {
                PopupExtensions.SetSize(this, VirtualView);
                PopupExtensions.SetLayout(this, VirtualView);
            }
        });
        base.ViewWillTransitionToSize(toSize, coordinator);
    }

    void SetDimmingBackgroundEffect()
    {
        if (ViewController?.View is UIView view)
        {
            OverlayView.Bounds = view.Bounds;
            OverlayView.Layer.RemoveAllAnimations();
            OverlayView.Frame = view.Frame;
            OverlayView.BackgroundColor = UIColor.Black.ColorWithAlpha(0.4f);
            view.AddSubview(OverlayView);
        }
    }

Before the MapOnClosed method of PopupHandler.macios.cs is called, the ViewWillDisapper 
method of MauiPopup.macios.cs is called, but the OverlayView has already been disposed.

The current implementation of the MapOnClosed method looks like this:

[src\CommunityToolkit.Maui.Core\Handlers\Popup\PopupHandler.macios.cs]

    public static async void MapOnClosed(PopupHandler handler, IPopup view, object? result)
    {
        var pc = handler.PlatformView.PresentationController;
        if (pc is not null)
        {
            if (pc.PresentedViewController is UIViewController pvc)
            {
                await pvc.DismissViewControllerAsync(true);
            }
        }

        view.HandlerCompleteTCS.TrySetResult();

        handler.DisconnectHandler(handler.PlatformView);
    }

The page that displays the Popup creates an instance of the Popup each time it is displayed.

In the display source of the Popup, an instance of the Popup is generated each time the Popup 
is displayed, but the OverlayView has already been destroyed when the Popup is closed by 
tapping the outside of the Popup. The OverlayView was destroyed before the OnMapClosed method was called.

Reply:
Thanks! Do you think we should to reinstantiate `OverlayView` if it is null somewhere in `Popup.Show()` / `Popup.ShowAsync()` code path?

This is just an honest question, not a suggestion as I haven't reviewed the code. @TRybina132 has volunteered to lead the review this PR. I just chimed in since I noticed that my PR adding `MemoryAnalysis` broke your PR.

Reply:
> Thanks! Do you think we should to reinstantiate OverlayView if it is null somewhere in Popup.Show() / Popup.ShowAsync() code path?

I added an OverlayView to show a dim background behind the Popup.
If you create a new OverlayView just because the OverlayView was destroyed while the overlay 
was being displayed, the OverlayView that was already placed on the screen will remain on 
the screen without being removed. I think it's not a solution.

Below is the verification video.

https://github.com/CommunityToolkit/Maui/assets/125236133/f0335e0c-ba45-4943-b057-e5b52976213c

I understand that this is not a review. Thank you for your time.
If I find out anything, I'll post it here.

Reply:
When WeakReference is introduced, Target becomes null irregularly.
It's a movement that can't be explained without the garbage collector working, 
but I don't know if the garbage collector actually worked or not.
Is it because timing that is not referenced always occurs somewhere ...

Reply:
@brminnick , I would like to wait until a concrete implementation method is established.
Although unrelated to this case, the iOS Popup cannot use instances to call the ShowPopupAsync method. 
I will raise an issue on this matter tomorrow. When the CreateControl method is called a second time, the Popup's Content has already been set to the ContentPage's Content in the first call, so the second time there is a warning.
I think the ViewWillAppear method is better for calling the SetDimmingBackgroundEffect method.
I think that the processing of the SetView method also needs to be reviewed. In any case, it is not implemented to reuse Popup instances.

Reply:
I am generating an overlay View when the Popup is displayed, but I want to delete this View when the 
Popup is closed, so I temporarily hold it in a class variable. The view for this overlay should be 
kept as a WeakReference, but no one will reference it long-term during the Popup's lifetime.
Therefore, it is immediately discarded by the garbage collector.

In this PR, we only need to get the view for the overlay when the Popup is closed and remove it 
properly, so we propose the following alternatives.

[src\CommunityToolkit.Maui.Core\Views\Popup\MauiPopup.macios.cs]

    UIView GetOverlayView(UIView view)
    {
        var overlayViewTag = new IntPtr(38483);
        var overlayView = view.Subviews.AsEnumerable()
            .Where(
                x => x.Tag == overlayViewTag
            )
            .FirstOrDefault();
        if (overlayView == null)
        {
            overlayView = new UIView();
            overlayView.Tag = overlayViewTag;
        }

        return overlayView;
    }

    void SetDimmingBackgroundEffect()
    {
        if (ViewController?.View is UIView view)
        {
            var overlayView = GetOverlayView(view);
            overlayView.Bounds = view.Bounds;
            overlayView.Layer.RemoveAllAnimations();
            overlayView.Frame = view.Frame;
            overlayView.BackgroundColor = UIColor.Black.ColorWithAlpha(0.4f);
            view.AddSubview(overlayView);
        }
    }

    public override void ViewWillTransitionToSize(CGSize toSize, IUIViewControllerTransitionCoordinator coordinator)
    {
        coordinator.AnimateAlongsideTransition((IUIViewControllerTransitionCoordinatorContext obj) =>
        {
            // Before screen rotate
            if (ViewController?.View is UIView view)
            {
                var overlayView = GetOverlayView(view);
                overlayView.Frame = new CGRect(view.Frame.X, view.Frame.Y, view.Frame.Width, view.Frame.Height);
            }
        }, (IUIViewControllerTransitionCoordinatorContext obj) =>
        {
            // After screen rotate
            if (VirtualView is not null)
            {
                PopupExtensions.SetSize(this, VirtualView);
                PopupExtensions.SetLayout(this, VirtualView);
            }
        });
        base.ViewWillTransitionToSize(toSize, coordinator);
    }

    public override void ViewWillDisappear(bool animated)
    {
        if (ViewController?.View is UIView view)
        {
            var overlayView = GetOverlayView(view);
            overlayView.RemoveFromSuperview();
            overlayView.Dispose();
        }
        base.ViewWillDisappear(animated);
    }

With this implementation, you don't need to keep the overlay view in a class variable.
I am using Tag to search Subviews for overlay view. 
I remembered that Tag was used in the implementation below and thought about it.

[src\CommunityToolkit.Maui.Core\Platform\StatusBar\StatusBar.ios.cs]

It would have been nice if Tag could be set to a string instead of a number, 
but it's a pity that it only has numbers... The value set for Tag is the value of 
StatusBar's Tag plus 1, and has no deep meaning.